cannon/cmd/run.go

```diff
@@ -64,6 +64,10 @@ var (
 		Value:    new(StepMatcherFlag),
 		Required: false,
 	}
+	RunStopAtPreimageFlag = &cli.BoolFlag{
+		Name:  "stop-at-preimage",
+		Usage: "stop at the first preimage request",
+	}
```

Comment:
This will always hit reading a local preimage key. It might be useful to be able to specify either the preimage type or even a specific preimage hash to stop at (which would make testing large preimages easier).

Reply:
Good idea. You can avoid the local preimage by iteratively updating the input prestate. Tho specifying the preimage type for the short-circuit avoids the need to do that.

Reply:
Added the option to specify both the preimage type and a specific preimage hash to stop at.

Reply:
Ah yeah, I hadn't thought about the start at param being able to skip ahead. Definitely easier to just specify what you want to stop at and let cannon sort it out though.